### PR TITLE
Feature/create synapse script

### PR DESCRIPTION
**Pull request by kingfardthe1st**
Updated shell scripts to deploy Azure resources and run synapse pipelines

### CODE REVIEW

File: streaming/synapse/runners/verify-synapse-pipeline.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Strict mode, fail on any error
+set -euo pipefail
+
+wait_for_completion="$1"
+echo $wait_for_completion
+SYNAPSE_SPARKPOOL="sasesssparkpool"
+AVRO_TO_DELTA_PIPELINE_NAME="blob-avro-to-delta-synapse"
+VERIFY_PIPELINE_NAME="verify-avro-to-delta"
```
Comment (butlfpaul) [inline streaming/synapse/runners/verify-synapse-pipeline.sh]: I think we should either rename this script or rename this variable. I think it should mimic the start-databricks-pipline script where the start script is generic and the parameters/variables are passed to this script from another script

File: streaming/synapse/runners/verify-synapse-pipeline.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    if [ $status != "Succeeded" ]; then
+        echo "Unsuccessful run"
+    fi
+}
+BLOB_BASE_PATH="streamingatscale/capture/$eventHubsNamespace/$eventHubName"
+tmp=$(mktemp)
+jq --arg a "${BLOB_BASE_PATH}" '.folder_path = $a' $DELTA_PIPELINE_PARAMETER_FILE > "$tmp" && mv "$tmp" $TEMP_PIPELINE_PARAMETER_FILE
+echo "Starting Avro to Delta Pipeline"
+DELTA_TEMP_RUN_ID=$(az synapse pipeline create-run --workspace-name $SYNAPSE_WORKSPACE \
```
Comment (butlfpaul) [inline streaming/synapse/runners/verify-synapse-pipeline.sh]: I think we should rename to just RUN_ID

File: streaming/synapse/runners/start-synapse-pipeline-trigger.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+PIPELINE_NAME="blob-avro-to-delta-synapse"
```
Comment (butlfpaul) [inline streaming/synapse/runners/start-synapse-pipeline-trigger.sh]: I'm wondering if this is the best place to put the start pipeline script

File: streaming/synapse/runners/verify-synapse-pipeline.sh (hunk 2 above)
Comment (kingfardthe1st) [reply streaming/synapse/runners/verify-synapse-pipeline.sh]: There are two different RUN_ID's, one for the verify pipeline, one for the avro-to-delta pipeline and I think we should differentiate the two.

Comment (kingfardthe1st) [reply streaming/synapse/runners/verify-synapse-pipeline.sh]: I've refactored the create-pipeline into a reusable script `run-synapse-pipeline.sh` which is used to create and run both the avro-to-delta and the verify pipelines just like the databricks sample.

File: streaming/synapse/runners/verify-synapse-pipeline.sh (hunk 1 above)
Comment (kingfardthe1st) [reply streaming/synapse/runners/verify-synapse-pipeline.sh]: I have renamed the script to verify-delta and refactored the pipeline creation and running steps into the `run-synapse-pipeline.sh` script. We just invoke the `create_pipeline_run` function with the appropriate parameters to run the pipelines.

File: streaming/synapse/runners/start-synapse-pipeline-trigger.sh (hunk 3 above)
Comment (kingfardthe1st) [reply streaming/synapse/runners/start-synapse-pipeline-trigger.sh]: I have a script that creates and run a triggered pipeline. This is called in the Processing Step.

**Review by butlfrazp — APPROVED** [review]

**Pull request merged**